testing/examples/test_nested_pipelines.py

```diff
@@ -0,0 +1,10 @@
+import numpy as np
+
+from examples.nested_pipelines import main
+
+
+def test_nested_pipelines():
+    n_components = main()
+
+    assert isinstance(n_components, np.ndarray)
+    assert n_components.shape == (2, 4)
```

Comment:
asserts could be inside the main again. The assert of the type is useless if we assert on the shape already.

Reply:
yeah, I figured that would be better while doing it. was already moving the asserts inside the examples